File: task/src/main/java/com/goexp/galgame/data/task/handler/DownloadBrandHandler.java
package com.goexp.galgame.data.task.handler;

import com.goexp.galgame.data.piplline.core.Message;
import com.goexp.galgame.data.piplline.core.MessageQueueProxy;
import com.goexp.galgame.data.piplline.handler.DefaultMessageHandler;
import com.goexp.galgame.data.task.client.GetChu;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DownloadBrandHandler extends DefaultMessageHandler<Integer> {

    private final Logger logger = LoggerFactory.getLogger(DownloadBrandHandler.class);

    @Override
    public void process(Message<Integer> message, MessageQueueProxy<Message> msgQueue) {

        var id = message.entity;
        logger.info("Down:{}", id);

        try {
            GetChu.BrandService.download(id);
        } catch (Exception e) {
            logger.error("ReDown:{}", id);
            msgQueue.offer(new Message<>(99, id));
            e.printStackTrace();
        }

        msgQueue.offer(new Message<>(MesType.Brand, id));
    }
}

File: task/src/main/java/com/goexp/galgame/data/task/handler/DownloadBrandHandler.scala
package com.goexp.galgame.data.task.handler

import com.goexp.galgame.data.piplline.core.{Message, MessageQueueProxy}
import com.goexp.galgame.data.piplline.handler.DefaultMessageHandler
import com.goexp.galgame.data.task.client.GetChu
import org.slf4j.LoggerFactory

class DownloadBrandHandler extends DefaultMessageHandler[Int] {
  final private val logger = LoggerFactory.getLogger(classOf[DownloadBrandHandler])

  override def process(message: Message[Int], msgQueue: MessageQueueProxy[Message[_]]): Unit = {
    val id = message.entity
    logger.info("Down:{}", id)
    try GetChu.BrandService.download(id)
    catch {
      case e: Exception =>
        logger.error("ReDown:{}", id)
        msgQueue.offer(new Message[Int](99, id))
    }
    msgQueue.offer(new Message[Int](MesType.Brand, id))
  }
}